webtransport/echo.sub.any.js

```diff
@@ -0,0 +1,42 @@
+// META: global=window,worker
```

Comment:
Moved the test in `infrastructure`. I added expectations but specifying `expected: FAIL` doesn't seem to work. I'm having trouble to understand why. Here is an example output:

```
 0:09.02 TEST_START: /infrastructure/server/webtransport-h3.sub.any.worker.html
 0:09.02 DEBUG Got command: 'log'
 0:09.02 DEBUG Executor setup done
 0:09.21 DEBUG Got command: 'log'
 0:09.21 DEBUG Got async callback: complete
 0:09.24 DEBUG Got command: 'test_ended'
 0:09.24 DEBUG Unexpected count in this thread 1
 0:09.24 DEBUG Unexpected count in this thread 2
 0:09.24 TEST_END: Test OK, expected FAIL. Subtests passed 0/1. Unexpected 1
FAIL WebTransport server should be running and should handle a bidirectional stream - promise_test: Unhandled rejection with value: object "ReferenceError: WebTransport is not defined"
Error
    at get_stack (http://web-platform.test:8000/resources/testharness.js:3780:21)
    at new AssertionError (http://web-platform.test:8000/resources/testharness.js:3773:22)
    at assert (http://web-platform.test:8000/resources/testharness.js:3763:19)
    at Test.<anonymous> (http://web-platform.test:8000/resources/testharness.js:664:29)
    at Test.step (http://web-platform.test:8000/resources/testharness.js:2087:25)
    at http://web-platform.test:8000/resources/testharness.js:2114:35
UNEXPECTED-OK /infrastructure/server/webtransport-h3.sub.any.worker.html
```

Note:  The error "ReferenceError: WebTransport is not defined" is expected since WebTransport isn't exposed by default. The error doesn't actually matter. My question is why expectations (.ini) don't seem to be working.

Reply:
It looks like you're setting the overall test expectation to `FAIL` which doesn't make sense (for testharness tests the overall test is `OK` unless there's an unhandled exception, in which case it's `ERROR` or a crash in which case its `CRASH`). You need to set the status specific to the subtest that actually fails, something like:
```
[webtransport-h3.sub.any.worker.html]
  [WebTransport server should be running and should handle a bidirectional stream]
    expected: FAIL
```

Reply:
Thank you for the detailed explanation! Looks like the test passes/fails in various ways (depending on browsers and settings) and I ended up adding many expectations. I feel I'm doing it wrong but couldn't figure out a better workaround.

I think we can remove these expectations once the server is enabled and https://crbug.com/1237781 is fixed.